bindings/c/src/lib.rs

```diff
@@ -0,0 +1,202 @@
+use css_inline::{CSSInliner, InlineError, InlineOptions, Url};
+use libc::{c_char, size_t};
+use std::borrow::Cow;
+use std::ffi::CStr;
+use std::io::Write;
+use std::ptr::null;
+
+/// Result of CSS inlining operations
+#[repr(C)]
+pub enum CssResult {
+    /// No error.
+    Ok,
+    /// Missing a stylesheet file.
+    MissingStylesheet,
+    /// When loading a remote stylesheet, the file is not available.
+    RemoteStylesheetNotAvailable,
+    /// Error in the IO layer. This error also happens when the output array is too small to fit
+    /// the inlined CSS.
+    IoError,
+    /// Error while parsing the CSS.
+    InternalSelectorParseError,
+    /// options pointer is null.
+    NullOptions,
+    /// Invalid base_url parameter.
+    InvalidUrl,
+    /// Invalid extra_css parameter.
+    InvalidExtraCss,
+    /// input string not in UTF-8.
+    InvalidInputString,
+}
+
+// must be public because the impl From<&CssInlinerOptions> for InlineOptions would leak this type
+/// Error to convert to CssResult later
+/// cbindgen:ignore
+pub enum InlineOptionsError {
+    InvalidUrl,
+    InvalidExtraCss,
+}
+
+/// Configuration options for CSS inlining process.
+#[repr(C)]
+pub struct CssInlinerOptions {
+    /// Keep "style" tags after inlining.
+    pub keep_style_tags: bool,
+    /// Keep "link" tags after inlining.
+    pub keep_link_tags: bool,
+    /// Used for loading external stylesheets via relative URLs.
+    pub base_url: *const c_char,
+    /// Whether remote stylesheets should be loaded or not.
+    pub load_remote_stylesheets: bool,
```

Comment:
As we use `repr(C)` would it be better to put `bool` a bit above? for the alignment reasons (the resulting `CssInlinerOptions` would be smaller)

Reply:
Didn't notice this.